### ISSUE TITLE
Milestones are not informative

### ISSUE THREAD

**Issue by tommck** [angular/router#254]
If you look at the milestones page on here, you see:

0.7.0 past due by 4 days - 0% complete, 16 open, 0 closed
0.5.3 past due by 3 days - 0% complete, 15 open, 0 closed
0.6.0 due april 14th - 13% complete, 25 open, 4 closed

That's a scary set of statistics to look at when you're hoping to find a roadmap for software.

Is there any chance this stuff can have some more realistic information in it?

**Comment by btford** [comment]
@tommck – totes agree.

There's some work being done on angular/angular to make the router work in Angular 2, which is why these milestones seem to be slipping.

These milestones will be more meaningful in about a week when those changes land. And the router will also work in Angular 2!

**Comment by Simon-Li** [comment]
Expect to see the new router working soon for A2, so trying to test on a determined branch is  always better than thinking across two options.

**Comment by tommck** [comment]
I was in the middle of trying to port a ui-router based app to the new router, but things just don't seem to be ready for the move yet.  I've been chomping at the bit for this forever, especially because I've done Durandal development before and really liked the router there.

Well, I'm waiting with bated breath :)

**Comment by Avien** [comment]
I am too considering to replace the ui router in the new router especially because I am starting a new big project in my company ,and already implemented ES6 syntax & new module system using traceur
However I am trying to get feedback from people who tried it already to decide on this change but after seeing your comment I already feel uncomfortable using it.
But on the other hand it would be a shame not using it now since this project will get bigger and it would be difficult doing it then
Brian, when do you suppose it will be good enough to start using it? in a week, when merged with A2?

**Comment by kpgarrod** [comment]
I also tried to implement the new router I  place of ui-router in a project
and back-tracked. However, I do feel that most of the work can be done in a
way that will make it very simple to migrate when the new router is ready,
so I don't think it's such a big deal.

Just implementing States with templateUrl, controller and controllerAs,
with the new router standard names takes care of most of the work.

I  am just including  canActivate and activate functions, with a temporary
canActivate().then(activate) in the start of the controller code.

It seems to me there will be few code changes when I switch,  unless I'm
missing something.

**Comment by Avien** [comment]
Yes you are right, its not such a big deal, ui router encourage you to create components by feature in the same folder anyway.
My only concern now is the authentication I've already implemented in the $stateChangeStart.
I will need to port it to the canActivate hook once ready

**Comment by tommck** [comment]
@kpgarrod could you share some implementation details of how you wired up the canActivate stuff in ui-router?  I'm intrigued.

**Comment by tommck** [comment]
@Avien, yeah you'll have to move that to a canActivate, but There will *have* to be some smooth way to handle that. I'd expect they'd have a nice auth example soon

**Comment by tommck** [comment]
@kpgarrod did you add the canActivate and activate methods on the state and not the controller?  I'm trying to figure out how they could be on the controller, but not coming up with anything that would make sense to me

**Comment by kpgarrod** [comment]
@tom, on the controller. sorry I haven't found the time to respond in more
detail. Just following John Papa's style-guide (except I'm using
controllerName in place of vm), there is an 'activate' function up-front in
the controller. I added a canActivate function, which returns a promise and
replaced the 'activate()' function call with
'canActivate().then(activate)'. I'm not sure that it would solve your
problem, but it seems to do what I need. Sorry I can't paste an example
right now, gtg.

**Comment by tommck** [comment]
But that stuff is called inside the Controller.  how does that stop the view from displaying if canActivate() resolves to false?

**Comment by krimple** [comment]
My biggest concern is that it's May 5th, and the last update on npm is from March 30th and there is nothing here that I've found describing the current plans. I had to find out about recent changes on a youtube video https://www.youtube.com/watch?v=sfV0ApyN5fM and the daily standup doc stopped being updated as of a few weeks ago.

Also, now it's being built on the angular 2 repository, and so I have no idea where to pull the latest router build from without building all of Angular 2.  npm and bower updates stopped March 30th. Should Angular 1.x developers just wait on progress of this router until an announcement and not attempt working on it? I'd love to help by kicking the tires and filing issues, but I feel it's not ready.

I just feel like we need a more realistic status update on the progress of the router for Angular 1.x developers.  If you're focusing more on 2.x, fine, but at ng-conf we were all told that ng-new-router would be delivered with 1.4 which was just about ready to go. Now it's almost two months later.

Again not complaining, just laying out what I have to explain when people ask me the progress of Angular 1.4 and the new router. I'd like some sort of content to point to instead of trying to tell the story from my perspective which is probably somewhat wrong.

**Comment by kpgarrod** [comment]
I also watched the video and would like to try it again with 1.4. Is the
version shown in the video documented somewhere?

**Comment by rrajewski** [comment]
+1
